docs/how_to_contribute/math/how_to_add_prime_field.md

```diff
@@ -0,0 +1,114 @@
+# How to add Prime Field
+
+Follow this guide to add a new Prime Field.
+
+_Note_: We have our own development conventions. Please read the [conventions docs](/docs/how_to_contribute/conventions.md) before contributing.
+
+## 1. Add `BUILD.bazel` file
+
+Begin by creating a directory named `<new_prime_field>_prime` in `/tachyon/math/finite_field/`. Add a `BUILD.bazel` into this directory. Note that once parameters are added to `BUILD.bazel`, Bazel will automatically generate the Prime Field code based on these parameters.
+
+Choose among `generate_prime_fields()`, `generate_fft_prime_fields()`, or `generate_large_fft_prime_fields()` for code generation, depending on the Prime Field type. For more information, refer to the [`prime_field_generator`](/tachyon/math/finite_fields/generator/prime_field_generator/build_defs.bzl).
+
+Additionally, it can be optimized using `polygon_zkevm_backend`. To utilize this, include `<field_type>_fail.h` and `<field_type>_fail.cc`.
+
+For instance, to implement a FFT Prime Field, create a directory (`/tachyon/math/finite_field/<new_prime_field>`) and add a `BUILD.bazel` file as shown below:
+
+```bazel
+# /tachyon/math/finite_field/<new_prime_field>/BUILD.bazel
+
+load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
+load("//bazel:tachyon.bzl", "if_polygon_zkevm_backend")
+load("//bazel:tachyon_cc.bzl", "tachyon_cc_library")
+load(
+    "//tachyon/math/finite_fields/generator/prime_field_generator:build_defs.bzl",
+    "SUBGROUP_GENERATOR",
+    "generate_fft_prime_fields",
+)
+
+package(default_visibility = ["//visibility:public"])
+
+string_flag(
+    name = "fr_" + SUBGROUP_GENERATOR,
+    build_setting_default = "{subgroup_generator}", # input Subgroup generator value
+)
+
+generate_fft_prime_fields(
+    name = "new_prime_field", # FIX
+    class_name = "NewPrimeField", # FIX
+    hdr_include_override = """#if defined(TACHYON_POLYGON_ZKEVM_BACKEND)
+#include "tachyon/math/finite_fields/<new_prime_field>_prime/prime_field_<new_prime_field>.h" # FIX
+#include "tachyon/build/build_config.h"
+#else
+#include "tachyon/math/finite_fields/prime_field.h"
+#endif  // defined(TACHYON_POLYGON_ZKEVM_BACKEND)""",
+    modulus = "{moulus_value}", # input modulus value
+    namespace = "tachyon::math",
+    special_prime_override = """#if defined(TACHYON_POLYGON_ZKEVM_BACKEND)
+#if ARCH_CPU_X86_64
+  constexpr static bool kIsSpecialPrime = true;
+  constexpr static bool kIsNewPrimeField = true; # FIX
+#else
+  #error Polygon NewPrimeField backend is only supported on x86_64 platform.
+#endif
+#else
+  constexpr static bool kIsSpecialPrime = false;
+#endif""",
+    subgroup_generator = ":fr_" + SUBGROUP_GENERATOR,
+    deps = if_polygon_zkevm_backend([
+        ":prime_field_<new_prime_field>", # FIX
```

Comment:
what is the "FIX"?

Reply:
supposed to be an indication of lines you should change :thinking: maybe should think of a better way to do this hmm...

Reply:
I just added a comment to notice users that "this line should be changed as your prime field name".

Reply:
hmm maybe `TODO: change as needed` might be better :thinking:

Reply:
I think it can be just removed cuz I abstracted the name like <new_prime_field>

Reply:
Hmm i think it would be better with a comment of some sort to help devs sort through what they have to do more easily.